ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/data/PathUtils.kt

```diff
@@ -0,0 +1,83 @@
+package com.revenuecat.purchases.ui.revenuecatui.customercenter.data
+
+import com.revenuecat.purchases.customercenter.CustomerCenterConfigData
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.composables.SettingsButtonStyle
+
+/**
+ * Utility object for categorizing and filtering customer center help paths
+ */
+internal object PathUtils {
+
+    /**
+     * Determines if a path is a general action that should be shown regardless of specific purchase context
+     */
+    fun isGeneralPath(path: CustomerCenterConfigData.HelpPath): Boolean {
+        return when (path.type) {
+            CustomerCenterConfigData.HelpPath.PathType.MISSING_PURCHASE,
+            CustomerCenterConfigData.HelpPath.PathType.CUSTOM_URL,
+            CustomerCenterConfigData.HelpPath.PathType.UNKNOWN,
+            -> true
+            CustomerCenterConfigData.HelpPath.PathType.CANCEL,
+            CustomerCenterConfigData.HelpPath.PathType.REFUND_REQUEST,
+            CustomerCenterConfigData.HelpPath.PathType.CHANGE_PLANS,
+            -> false
+        }
+    }
+
+    /**
+     * Determines if a path is a subscription-specific action
+     */
+    fun isSubscriptionSpecificPath(path: CustomerCenterConfigData.HelpPath): Boolean {
+        return when (path.type) {
+            CustomerCenterConfigData.HelpPath.PathType.CANCEL,
+            CustomerCenterConfigData.HelpPath.PathType.REFUND_REQUEST,
+            CustomerCenterConfigData.HelpPath.PathType.CHANGE_PLANS,
+            -> true
+            CustomerCenterConfigData.HelpPath.PathType.MISSING_PURCHASE,
+            CustomerCenterConfigData.HelpPath.PathType.CUSTOM_URL,
+            CustomerCenterConfigData.HelpPath.PathType.UNKNOWN,
+            -> false
+        }
+    }
```

Comment:
We're missing the purchase state specific paths, like not showing cancelled if already cancelled, right?

Reply:
yes, showing resuscribe if already cancelled. I wanted to address those smaller changes in separate PRs